### ISSUE TITLE
flash of text

### ISSUE THREAD

**Issue by ngokevin** [aframevr/aframe-site#377]
Any way to avoid the flash of text from the built-in font to the web font? I think this is to avoid invisible text. Perhaps Service Workers?

**Comment by cvan** [comment]
> Any way to avoid the flash of text from the built-in font to the web font?

Previously, all text on the page was invisible until the web fonts loaded. I added 
the [Font Face Observer](https://github.com/bramstein/fontfaceobserver) library to "fix" that. Here's the code: https://github.com/aframevr/aframe-site/blob/4b33b5e/themes/aframe/source/js/common.js#L17-L21

It's not a great solution, but the text is at least visible and readable, getting then swapped out for the text rendered with the web fonts.

The [Font Loading API](https://drafts.csswg.org/css-font-loading/) is being worked on to handle all these shenanigans.

Lately, I've been forgoing custom web fonts in favour of system fonts. It's fast and looks reasonably well everywhere, since the fonts are inconsistent with the usage of other text in other applications on the system.

### `sans-serif` family
```css
font-family: -apple-system, BlinkMacSystemFont,
    Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans,
    Droid Sans, Helvetica Neue, sans-serif;
```

### `monospace` family
```css
font-family: Consolas, Andale Mono, Monaco,
  Lucida Console, Liberation Mono, DejaVu Sans Mono,
  Bitstream Vera Sans Mono, Courier New, monospace;
```

Or you could use this cool trick: https://github.com/jonathantneal/system-font-css

If you really want to keep Fira, you can preload the font files. But keep in mind this will work for only Chrome, and Chrome supports WOFF2, so you need to add `<link rel="preload">` for only the `.woff2` font files.

```html
<link rel="preload" href="https://aframe.io/fonts/fira/woff2/FiraSans-Hair.woff2" as="font" type="font/woff2">
```

Not a great solution. And, Service Workers aren't available in Safari/WebKit yet, so the network request for the font files will still be a problem everywhere on iOS.

In summary, I'd recommend using system fonts.

**Comment by ngokevin** [comment]
Cool thanks, I'll play with system fonts some time. Fira is nice for the branding, but the flash is a bit janky.

**Comment by ngokevin** [comment]
removed the JS that caused the flash
